src/general.md

```diff
@@ -208,7 +208,63 @@ const branch = company.branch;
 
 ## Comments
 
-1. Use comments to add context or information that is not obvious from the code already. Don't use comments to explain the code itself - code (through proper naming) should be self-explanatory.
+1. Use code to express intent.
+
+   - Code should be the primary form of documentation.
+   - Instead of writing comments to explain _what_ the code does, structure and name the code so that it is self-explanatory.
+   - Good code reads like a narrative of the intention.
+   - Comments should be reserved for explaining _why_ something is done in a non-obvious way.
+
+   ```typescript
+   // bad
+   const list = {
+      // old items
+      oldItem1,
+      ...
+      // the current one
+      item1,
+      ...
+   };
+
+   // good
+   const oldItems = {
+      oldItem1,
+      ...
+   };
+   const list = {
+      ...oldItems,
+      item1,
+      ...
+   };
+   ```
+
+   ```typescript
+   // bad
+   // Open X page
+   await expect(page.locator('text="X link"')).toBeVisible();
+   await page.click('[data-autotest="x"]');
+   await expect(page.locator('text=X page title')).toBeVisible();
+
+   // good
+   await openXPage(page);
+   ```
```

Comment:
🍓  As an engineer, my first priority is always to think about how to express my thoughts through code

Reply:
> - Comments should be reserved for explaining _why_ something is done in a non-obvious way.

would be great to have an example